src/kdbindings/connection_handle.h

```diff
@@ -293,8 +296,11 @@ class ScopedConnection
 
     /**
      * When a ConnectionHandle is destructed it disconnects the connection it guards.
+     *
+     * @warning While this function is marked with noexcept, it *may* terminate the program
+     * if it is not possible to allocate memory or if mutex locking isn't possible.
      */
-    ~ScopedConnection()
+    ~ScopedConnection() noexcept
```

Comment:
Since destructors are implicitly `noexcept(true)` it's probably redundant to mark it as such (same with other instances of `~Foo() noexcept`)?
```suggestion
     * @warning While this function isn't marked as throwing, it *may* throw and terminate
     * the program if it is not possible to allocate memory or if mutex locking isn't
     * possible.
     */
    ~ScopedConnection()
```

On the other hand, we're then pretty explicit about our intention here and it doesn't hurt. Feel free to leave or change.

Reply:
I incorporated your suggested comments, but chose to keep the noexcept specifier.
It's more explicit, and I honestly had forgotten that throwing within a constructor terminates the program anyway, so this is a good way to document this behavior.

Reply:
Makes sense